pkg/runtime/embedded.go

```diff
@@ -69,17 +69,22 @@ func (re *Unknown) UnmarshalJSON(in []byte) error {
 		return errors.New("runtime.Unknown: UnmarshalJSON on nil pointer")
 	}
 	re.TypeMeta = TypeMeta{}
-	re.RawJSON = append(re.RawJSON[0:0], in...)
+	re.Raw = append(re.Raw[0:0], in...)
+	re.ContentEncoding = ""
+	re.ContentType = ContentTypeJSON
 	return nil
 }
 
 // Marshal may get called on pointers or values, so implement MarshalJSON on value.
 // http://stackoverflow.com/questions/21390979/custom-marshaljson-never-gets-called-in-go
 func (re Unknown) MarshalJSON() ([]byte, error) {
-	if re.RawJSON == nil {
+	if re.ContentType != ContentTypeJSON {
```

Comment:
It should be allowed for content type to be empty, at which point it would default to JSON (as per the proposal).  I think that logic has to be here (as well as other places).